pkg/cri/sbserver/sandbox_run.go

```diff
@@ -164,17 +164,33 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		return nil, fmt.Errorf("unable to save sandbox %q to store: %w", id, err)
 	}
 
+	ociRuntime, err := c.getSandboxRuntime(config, r.GetRuntimeHandler())
```

Comment:
Should we do something like this?
```
if shim.SupportsSandboxes() { // check whether there is TTRPC sandbox service
 return makeRemoteSandobx(shim.GetClient())
els {
  return usePodSandboxImplementation()
}
```

Reply:
I cannot get the shim instance as it's created during Controller.Start(), when the sandbox already have been decided.